src/core_plugins/kibana/server/lib/make_kql_usage_collector.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getUiSettingDefaults } from '../../ui_setting_defaults';
+import { get } from 'lodash';
+import Promise from 'bluebird';
+
+const uiSettingDefaults = getUiSettingDefaults();
+const defaultSearchQueryLanguageSetting = uiSettingDefaults['search:queryLanguage'].value;
+
+export function makeKQLUsageCollector(server) {
+  const kqlUsageCollector = server.usage.collectorSet.makeUsageCollector({
+    type: 'kql',
+    fetch: async callCluster => {
+      const [response, config] = await Promise.all([
+        callCluster('search', { index: '.kibana', q: 'type:kql-telemetry', ignore: [404] }),
```

Comment:
Looks like this query uses query string query, which uses scoring in Elasticsearch to rank the results. Maybe you don't need that? 

This would execute in filter context and not get a (negligible) performance hit from scoring:
```
{
  "query": {
    "bool": {
      "filter": {
        "term": {
          "type": "kql-telemetry"
        }
      }
    }
  }
}
```

Or, if you know the ID, you can do a `'get'` instead of a search. I'm not sure how reliable ID would be, it looks like it might be generated from some spaces magic.

Reply:
Good catch! I wrote this query before I decided to use a static ID, this can definitely turn into a `get` now. Namespaces won't be an issue since we don't specify a namespace when we create the doc. 

I also updated the query for the config doc to use a term query instead of a query string.